### PR TITLE
thingy53: samples and applications: Clean USB CDC related configs

### PR DESCRIPTION
**Pull request by mkapala-nordic**
Thingy:53 board in the NCS already enables logging over USB CDC by default.
There is no need anymore to explicitly enable most of USB CDC related options in Thingy:53 app configs.

Jira: NCSDK-18596

Signed-off-by: Mateusz Kapala <mateusz.kapala@nordicsemi.no>

### CODE REVIEW

**Review by kkasperczyk-no — COMMENTED** [review]
Could you apply the same changes as in matter_weather_station/.../prj.conf to the prj_factory_data.conf?

File: applications/machine_learning/src/modules/Kconfig.usb_state [hunk 1]
```diff
@@ -12,6 +12,14 @@ menuconfig ML_APP_USB_STATE
 	help
 	  Enable USB subsystem on application start.
 
+if ML_APP_USB_STATE
```
Comment (MarekPieta) [inline applications/machine_learning/src/modules/Kconfig.usb_state:15]: I think we should consider disabling the `CONFIG_BOARD_SERIAL_BACKEND_CDC_ACM` in Thingy:53 MCUboot configuration. Actually it should not be needed for the MCUboot

Comment (MarekPieta) [reply applications/machine_learning/src/modules/Kconfig.usb_state:15]: We also should definitely disable one of the USB CDC ACM instances in Thingy:53 MCUboot's DTS (after your changes you have 2 USB CDC instanced and one of them is unused)

Comment (MarekPieta) [reply applications/machine_learning/src/modules/Kconfig.usb_state:15]: On instance is added by Thingy:53 board and the second one comes from `usb.overlay` applied automatically in the `nrf/modules/mcuboot/CMakelists.txt` file

File: samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp.conf [hunk 2]
```diff
@@ -59,22 +59,15 @@ CONFIG_NORDIC_QSPI_NOR_FLASH_LAYOUT_PAGE_SIZE=4096
...
-CONFIG_USB_DEVICE_VID=0x1915
 CONFIG_USB_DEVICE_PID=0x530E
 CONFIG_USB_COMPOSITE_DEVICE=y
-CONFIG_USB_CDC_ACM=y
-CONFIG_USB_CDC_ACM_LOG_LEVEL_ERR=y
-CONFIG_USB_DEVICE_LOG_LEVEL_ERR=y
 
-CONFIG_LOG_BACKEND_UART=y
 CONFIG_LOG_PRINTK=y
 CONFIG_LOG_PROCESS_THREAD_STACK_SIZE=1024
```
Comment (MarekPieta) [inline samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp.conf:69]: Do we need to configure these explicitly? I think we should ensure that the USB CDC configuration would work out-of-the-box also for other samples (that rely on default values here).

Comment (MarekPieta) [reply samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp.conf:69]: I would opt for removing all of the unnecessary configurations related to logging or USB CDC ACM to simplify the user experience (if something is mandatory we should probably set it by default in Thingy:53 board definition files).

**Review by MarekPieta — CHANGES_REQUESTED** [review]

**Comment by nordicjm** [comment]
Needs to be rebased to include west.yml changes and changes in https://github.com/nrfconnect/sdk-nrf/pull/9999 which removes some files touched here

File: samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp.conf (hunk 2 above)
Comment (mkapala-nordic) [reply samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp.conf]: Removed all unnecessary configs

**Comment by mkapala-nordic** [comment]
> Could you apply the same changes as in matter_weather_station/.../prj.conf to the prj_factory_data.conf?

@kkasperczyk-no Of course, i must have missed them at the first time, fixed.

> Needs to be rebased to include west.yml changes and changes in #9999 which removes some files touched here

Rebased onto #9999.

File: applications/nrf5340_audio/src/modules/dfu_entry.c [hunk 3]
```diff
@@ -18,19 +18,6 @@
 #include "macros_common.h"
 #include "channel_assignment.h"
 
-#ifdef CONFIG_MCUMGR_CMD_OS_MGMT
```
[the remark this answers is not in the document]
Comment (mkapala-nordic) [reply applications/nrf5340_audio/src/modules/dfu_entry.c:21]: I talked with @kapi-no, @MarekPieta how we should proceed.
We've come to conclusion that this PR can be independently merged before #9999. 
I'll rebase back to main and drop all changes from mentioned PR.

File: applications/machine_learning/src/modules/Kconfig.usb_state (hunk 1 above)
Comment (mkapala-nordic) [reply applications/machine_learning/src/modules/Kconfig.usb_state:15]: Discussed offline, taken steps:
+ sdk-mcuboot noup change where `CONFIG_BOARD_SERIAL_BACKEND_CDC_ACM` is disabled in thingy board config https://github.com/nrfconnect/sdk-mcuboot/pull/237
+ sdk-zephyr noup change where `CONFIG_MCUBOOT_USB_SUPPORT` is disabled in thingy board config https://github.com/nrfconnect/sdk-zephyr/pull/1043

File: samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp.conf [hunk 4]
```diff
@@ -59,22 +59,13 @@ CONFIG_NORDIC_QSPI_NOR_FLASH_LAYOUT_PAGE_SIZE=4096
...
-CONFIG_USB_CDC_ACM=y
-CONFIG_USB_CDC_ACM_LOG_LEVEL_ERR=y
-CONFIG_USB_DEVICE_LOG_LEVEL_ERR=y
 
 CONFIG_LOG_BACKEND_UART=y
-CONFIG_LOG_PRINTK=y
-CONFIG_LOG_PROCESS_THREAD_STACK_SIZE=1024
-CONFIG_LOG_PROCESS_THREAD_STARTUP_DELAY_MS=4000
 CONFIG_USE_SEGGER_RTT=n
 CONFIG_LOG_BACKEND_RTT=n
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp.conf:70]: IIRC we could actually provide logs both over CDC ACM and RTT (you can double check if it has no side effects).

File: samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp.conf [hunk 5]
```diff
@@ -59,22 +59,13 @@ CONFIG_NORDIC_QSPI_NOR_FLASH_LAYOUT_PAGE_SIZE=4096
...
-CONFIG_USB_DEVICE_REMOTE_WAKEUP=n
 CONFIG_USB_DEVICE_PRODUCT="Thingy:53 Nordic UART Service sample"
-CONFIG_USB_DEVICE_VID=0x1915
 CONFIG_USB_DEVICE_PID=0x530E
 CONFIG_USB_COMPOSITE_DEVICE=y
-CONFIG_USB_CDC_ACM=y
-CONFIG_USB_CDC_ACM_LOG_LEVEL_ERR=y
-CONFIG_USB_DEVICE_LOG_LEVEL_ERR=y
 
 CONFIG_LOG_BACKEND_UART=y
```
Comment (MarekPieta) [inline samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp.conf:68]: Please comment on why this is still needed

**Review by MarekPieta — CHANGES_REQUESTED** [review]

File: samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp_ns.conf [hunk 6]
```diff
@@ -59,22 +59,13 @@ CONFIG_NORDIC_QSPI_NOR_FLASH_LAYOUT_PAGE_SIZE=4096
...
-CONFIG_USB_DEVICE_REMOTE_WAKEUP=n
 CONFIG_USB_DEVICE_PRODUCT="Thingy:53 Nordic UART Service sample"
-CONFIG_USB_DEVICE_VID=0x1915
 CONFIG_USB_DEVICE_PID=0x530E
 CONFIG_USB_COMPOSITE_DEVICE=y
-CONFIG_USB_CDC_ACM=y
-CONFIG_USB_CDC_ACM_LOG_LEVEL_ERR=y
-CONFIG_USB_DEVICE_LOG_LEVEL_ERR=y
 
 CONFIG_LOG_BACKEND_UART=y
```
[the remark this answers is not in the document]
Comment (mkapala-nordic) [reply samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp_ns.conf:69]: Added comment

File: samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp.conf (hunk 4 above)
[the remark this answers is not in the document]
Comment (mkapala-nordic) [reply samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp.conf]: Both backends can work simultaneously so i just left enabling UART backend here.

File: applications/machine_learning/src/modules/Kconfig.usb_state [hunk 7]
```diff
@@ -12,6 +12,14 @@ menuconfig ML_APP_USB_STATE
...
+
+# Let USB device initialization to the USB module
```
[the remark this answers is not in the document]
Comment (mkapala-nordic) [reply applications/machine_learning/src/modules/Kconfig.usb_state]: Changed comment to Kconfig help

**Review by kapi-no — APPROVED** [review]

**Review by MarekPieta — APPROVED** [review]

**Comment by mkapala-nordic** [comment]
Rebase to newest main

**Comment by mkapala-nordic** [comment]
Rebased + CI rerun to fix compliance check fail from sdk-zephyr PR

File: applications/zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj.conf [hunk 8]
```diff
@@ -11,15 +11,12 @@
...
-CONFIG_UART_CONSOLE=y
 CONFIG_ZIGBEE_WEATHER_STATION_LOG_LEVEL_INF=y
 
 # USB
-CONFIG_USB_DEVICE_STACK=y
-CONFIG_USB_DEVICE_REMOTE_WAKEUP=n
+CONFIG_USB_DEVICE_INITIALIZE_AT_BOOT=n
 CONFIG_USB_DEVICE_PRODUCT="Thingy 53: Zigbee Weather Station"
 CONFIG_USB_CDC_ACM_LOG_LEVEL_WRN=y
-CONFIG_USB_DEVICE_VID=0x1915
```
[the remark this answers is not in the document]
Comment (nordicjm) [reply applications/zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj.conf:22]: not addressed

[the remark this answers is not in the document]
Comment (mkapala-nordic) [reply applications/zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj.conf:22]: @adsz-nordic, please take a look at @MarekPieta comment

[the remark this answers is not in the document]
Comment (sebastiandraus) [reply applications/zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj.conf:22]: Should be fine to use common USB PID, previously default Zephyr's one was used.

**Review by sebastiandraus — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Comment by mkapala-nordic** [comment]
Updated manifest after merging sdk-zephyr and sdk-mcuboot related PRs

**Comment by mkapala-nordic** [comment]
Rebased due to manifest conflict

**Pull request merged**